### PR TITLE
Small improvements 

### PR DESCRIPTION
**Pull request by Totto16**
- [x] this fixes meson build errors ( adding sdl2_ttf dependency)
- [x] enable -Werror -Wall -Wextra -Wpedantic and fix the arising errors
- [x] enable ninja instead of make in CMake CI, so it's faster (~2 times)
- [x] actions suceed

### CODE REVIEW

File: src/bag.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@ bool Bag::s_seeded = false;
 
 Bag::Bag() : m_tetromino_sequence{} {
     if (!s_seeded) {
-        std::srand(std::time(nullptr));
+        std::srand((unsigned int) std::time(nullptr));
```
Comment (mgerhold) [inline src/bag.cpp:10]: Please no C-style casts :)
```suggestion
        std::srand(static_cast<unsigned int>(std::time(nullptr)));
```

File: src/bag.cpp [hunk 2]
```diff
@@ -17,12 +17,12 @@ Bag::Bag() : m_tetromino_sequence{} {
     }
 
     // fill in the sequence with random order
-    for (int i = 0; i < m_tetromino_sequence.size(); ++i) {
+    for (size_t i = 0; i < m_tetromino_sequence.size(); ++i) {
```
Comment (mgerhold) [inline src/bag.cpp:20]:
```suggestion
    for (std::size_t i = 0; i < m_tetromino_sequence.size(); ++i) {
```

File: src/bag.cpp [hunk 3]
```diff
@@ -17,12 +17,12 @@ Bag::Bag() : m_tetromino_sequence{} {
...
         bool type_is_okay = true;
         do {
             type_is_okay = true;
             m_tetromino_sequence[i] = get_random_tetromino_type();
-            for (int j = 0; j < i; ++j) {
+            for (size_t j = 0; j < i; ++j) {
```
Comment (mgerhold) [inline src/bag.cpp:25]:
```suggestion
            for (std::size_t j = 0; j < i; ++j) {
```

File: src/game_manager.hpp [hunk 4]
```diff
@@ -58,7 +58,7 @@ struct GameManager final {
     void refresh_preview();
     TetrominoType get_next_tetromino_type();
 
-    static double get_gravity_delay(int level) {
+    static double get_gravity_delay(size_t level) {
```
Comment (mgerhold) [inline src/game_manager.hpp:61]:
```suggestion
    static double get_gravity_delay(std::size_t level) {
```

File: src/tetris_application.hpp [hunk 5]
```diff
@@ -49,6 +49,7 @@ struct TetrisApplication : public Application {
     }
 
     void update(double delta_time) override {
+        (void)delta_time;
         m_game.update();
     }
```
Comment (mgerhold) [inline src/tetris_application.hpp:54]: Let's use `[[maybe_unused]]` or remove the name of the parameter all together.
```suggestion
    void update([[maybe_unused]] double delta_time) override {
        m_game.update();
    }
```

**Review by mgerhold — COMMENTED** [review]
Just some minor change requests. Do you want to change them or should I do it and you rebase after that?

File: src/tetris_application.hpp (hunk 5 above)
Comment (Totto16) [reply src/tetris_application.hpp:54]: Also to override the update function (that definitely need a delta_time) we need the parameter :sweat_smile:

Comment (Totto16) [reply src/tetris_application.hpp:54]: So the solution would be: `void update(double) override`

**Comment by Totto16** [comment]
> Just some minor change requests. Do you want to change them or should I do it and you rebase after that?

I now used the github interface for the chnages, that work, for the one with `[[maybe_unused]]` (that's not available in c++14)  I made a commit with the solution that looks like this: `void update(double) override`

**Comment by Totto16** [comment]
The action failing has nothing to do with the code, but rather a dependecy issue :sweat_smile: It should be fine merging anyways (or rerun the action)

**Review by mgerhold — APPROVED** [review]

**Comment by mgerhold** [comment]
> > Just some minor change requests. Do you want to change them or should I do it and you rebase after that?
> 
> I now used the github interface for the changes, that work, for the one with `[[maybe_unused]]` (that's not available in c++14) I made a commit with the solution that looks like this: `void update(double) override`

Perfect, thanks 👍
I also downloaded a local copy for my C++ training, so I think we can go ahead and increase the C++ standard to 23 the next time we do an update.

**Pull request merged**